### PR TITLE
Upgrade Anomalib Version

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
### Summary

Note: Depends on Anomalib version 0.5.1. Merge only after that release is created.

- This PR modifies the metadata used by the anomaly OpenVINO inference tasks.
- It also bumps anomalib version to 0.5.1

### Checklist
Tested locally for anomaly classification task

image

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: otx/algorithms/anomaly/adapters/anomalib/exportable_code/base.py
Comment (sovrasov) [inline otx/algorithms/anomaly/adapters/anomalib/exportable_code/base.py:26]: Are these changes backward compatible? Sometimes OTX updates break loading legacy projects when upgrading Geti version

Comment (ashwinvaidya17) [reply otx/algorithms/anomaly/adapters/anomalib/exportable_code/base.py:26]: Not really. Do Geti projects rely on exportable code?

Comment (sovrasov) [reply otx/algorithms/anomaly/adapters/anomalib/exportable_code/base.py:26]: No, they don't, but I mean compatibility in general. Geti saves ModelEntities in it's database as a part of a project. OTX should be compatible with ModelEntities from previous Geti versions. We already had an issue with missing extra configs for tiling, so this question is just a precaution. If you don't see any problems with Geti upgrade scenario, that's fine.

**Comment by samet-akcay** [comment]
PR checks fail complaining that there is no anomalib v0.5.1 🤣

**Review by djdameln — DISMISSED** [review]

**Review by samet-akcay — DISMISSED** [review]
Thanks a lot for the effort!

**Comment by jaegukhyun** [comment]
Can it solve known issues for anomaly such as unit-test issue, or performance drop after export, optimize and deply?

**Comment by ashwinvaidya17** [comment]
> Can it solve known issues for anomaly such as unit-test issue, or performance drop after export, optimize and deply?

I haven't tested it for the performance drop and the python10 issue. This just targets the latest anomalib and makes the corresponding changes to the API.

**Comment by jaegukhyun** [comment]
Sound good. Thanks

**Comment by jaegukhyun** [comment]
What is the target version of OTX? Is it OTX1.3.1 or OTX1.4.0

**Comment by ashwinvaidya17** [comment]
> What is the target version of OTX? Is it OTX1.3.1 or OTX1.4.0

I don't know. I just created a branch from develop

**Review by jaegukhyun — APPROVED** [review]

**Review by samet-akcay — APPROVED** [review]

File: otx/algorithms/anomaly/adapters/anomalib/exportable_code/base.py
Comment (ashwinvaidya17) [reply otx/algorithms/anomaly/adapters/anomalib/exportable_code/base.py:26]: I have a feeling that it might affect Geti upgrade. But maybe we will have to fix it only once. The advantage of having metadata as a json is that we can be flexible later on if keys are added or removed. Feel free to ping me if this breaks during Geti project upgrade.

**Comment by ashwinvaidya17** [comment]
I cannot merge this PR. Can one of the maintainers merge this?

**Pull request merged**
